src/observer/ingest_logic.py

```diff
@@ -416,21 +418,61 @@ def mkidx():
 def _download_versions(msid, latest_version):
     LOG.info(' %s versions to fetch' % latest_version)
     version_range = range(1, latest_version + 1)
-    lmap(lambda version: upsert_ajson(consume("articles/%s/versions/%s" % (msid, version))), version_range)
+
+    def fetch(version):
+        upsert_ajson(msid, version, models.LAX_AJSON, consume("articles/%s/versions/%s" % (msid, version)))
+    lmap(fetch, version_range)
 
 def download_article_versions(msid):
-    "loads *all* versions of given article from the api"
+    "loads *all* versions of given article via API"
     resp = consume("articles/%s/versions" % msid)
-    return _download_versions(msid, len(resp["versions"]))
+    _download_versions(msid, len(resp["versions"]))
 
 def download_all_article_versions():
-    "loads *all* versions of *all* articles from the api"
-    msid_ver_idx = mkidx()
+    "loads *all* versions of *all* articles via API"
+    msid_ver_idx = mkidx() # urgh. this sucks. lax needs a /summary endpoint too
     LOG.info("%s articles to fetch" % len(msid_ver_idx))
     idx = sorted(msid_ver_idx.items(), key=lambda x: x[0], reverse=True)
     for msid, latest_version in idx:
         _download_versions(msid, latest_version)
 
+#
+# metrics data
+#
+
+def _upsert_metrics_ajson(data):
+    version = None
+    if utils.byte_length(data['msid']) > 8: # big ints in sqlite3 are 64 bits/8 bytes large
+        LOG.error("bad data encountered, cannot store msid: %s", data['msid'])
```

Comment:
`too big id encountered`?

Reply:
we can handle regular msid and testing msid, anything else is bad data. being specific here would be misleading and unhelpful. The given msid will help find the bad article in metrics`.